Drivers/Src/MotorController.c

```diff
@@ -89,36 +178,66 @@ ErrorStatus MotorController_Read(CANbuff *message, car_state_t *car){
             // If we're reading the output from the Motor Status command (0x241) then 
             // Check the status bits we care about and set flags accordingly
             case MOTOR_STATUS: {
-                if(MASK_MOTOR_TEMP_ERR & firstSum)
-                {
-                    car->MotorErrorCode.motorTempErr = ON;
-                }
-
-                if(MASK_SS_ERR & firstSum)
-                {
-                    car->MotorErrorCode.slipSpeedErr = ON;
-                }
-
-                if(MASK_CC_ERR & firstSum)
-                {
-                    car->MotorErrorCode.CCVelocityErr = ON;
-                }
-
-                if(MASK_OVER_SPEED_ERR & firstSum)
-                {
-                    car->MotorErrorCode.overSpeedErr = ON;
-                }
+                if(MASK_MOTOR_TEMP_ERR & firstSum) Motor_FaultBitmap |= T_TEMP_ERR;
+
+                if(MASK_SS_ERR & firstSum) Motor_FaultBitmap |= T_CC_VEL_ERR;
+
+                if(MASK_CC_ERR & firstSum) Motor_FaultBitmap |= T_SLIP_SPEED_ERR;
+
+                if(MASK_OVER_SPEED_ERR & firstSum) Motor_FaultBitmap |= T_OVER_SPEED_ERR;
+                
+                assertTritiumError(Motor_FaultBitmap);
+                
                 break;
             }
             case MOTOR_DRIVE: {
                 convert.n = secondSum;
-                car->CurrentVelocity = convert.f;
+
+                OSMutexPend(&MotorController_VelocityMutex,
+				0,
+				OS_OPT_PEND_BLOCKING,
+				&ts,
+				&err);
+	            assertOSError(0, err);
+
+                CurrentVelocity = convert.f;
```

Comment:
This is already atomic. You don't need a mutex for this

Reply:
Should we leave it just in case we have an update thread and a read thread? Or should I just remove it anyways. I don't think it will hurt to leave it in there.